include/graph_generator.h

```diff
@@ -61,9 +61,13 @@ namespace detail {
 		using side_effect_map = std::unordered_map<host_object_id, command_id>;
 
 		struct per_node_data {
-			// An "init command" is used as the last writer for host-initialized buffers.
+			// The most recent horizon command. Depends on the previous execution front and will become the epoch_for_new_tasks once the next horizon is
+			// generated.
+			std::optional<command_id> current_horizon;
+			// The active epoch command is used as the last writer for host-initialized buffers.
 			// This is useful so we can correctly generate anti-dependencies onto commands that read host-initialized buffers.
-			command_id current_init_cid;
+			// To ensure correct ordering, all commands that have no other true-dependencies depend on this command.
+			command_id epoch_for_new_commands;
```

Comment:
Since `per_node_data` does not maintain any other invariants, I've moved the method outside the struct into `graph_generator` (where all other mutators of `per_node_data` live).